ai21/models/ai21_base_model.py

```diff
@@ -0,0 +1,45 @@
+import re
+from typing import Any, Dict
+
+from pydantic import BaseModel, ConfigDict
+
+from ai21.models.pydantic_compatibility import IS_PYDANTIC_V2
+
+if IS_PYDANTIC_V2:
+    from pydantic.alias_generators import to_camel
+
+
+def to_camel_case(snake: str) -> str:
+    if re.match("^[a-z]+[A-Za-z0-9]*$", snake) and not re.search(r"\d[a-z]", snake):
+        return snake
+
+    pascal = re.sub("([0-9A-Za-z])_(?=[0-9A-Z])", lambda m: m.group(1), snake.title())
+    return re.sub("(^_*[A-Z])", lambda m: m.group(1).lower(), pascal)
+
+
+class AI21BaseModel(BaseModel):
+    if IS_PYDANTIC_V2:
+        model_config = ConfigDict(
+            alias_generator=to_camel,
+            populate_by_name=True,
+            protected_namespaces=(),
+        )
+    else:
+
+        class Config:
+            alias_generator = to_camel_case
+            allow_population_by_field_name = True
+
+    def to_dict(self) -> Dict[str, Any]:
```

Comment:
Perhaps we should raise a deprecation warning here and have people move to use pydantic native methods

Reply:
I'm going to commit a change to these functions, where according to the pydantic version I use either `model_dump` (v2) or `dict` (v1), and the equivalents for json, parse_obj, parse_raw. I was thinking to use them within our internal code, since they already do the logic of directing to the relevant function by version. I think that as long as we're still supporting both pydantic v1 and v2, this function is useful and we can wait with the deprecation notice.
WDYT?
cc: @pazshalev

Reply:
Discussed as we need to have the user get the deprecation warning + the strike out on the ~to_dict~

Reply:
Done